src/deepsparse/benchmark/benchmark_model.py

```diff
@@ -357,6 +359,14 @@ def benchmark_model(
 
     orig_model_path = model_path
     model_path = model_to_path(model_path)
+
+    if has_model_kv_cache(model_path):
+        _LOGGER.info(
+            "Found model that contains KV cache support. "
+            "Benchmarking the autoregressive model."
+        )
+        model_path = assert_model_sequence_length_one(model_path)
```

Comment:
we need to fail if the benchmark batch size is >1 in this case. right now it's not an assertion, just blanket overwriting the kv cache dims to bs 1

Reply:
also check if the engine supports bs > 1 on the kv cache dims. if so, we probably don't need this at all as the engine will update the dims on their own

Reply:
it would be good to have a way to pipe the sequence length through here so we can control varying that through this useful utility